src/components/dialogs/two-windings-transformer/ratio-tap-changer-pane.js

```diff
@@ -0,0 +1,535 @@
+    const editableCellRender = useCallback(
+        (rowData) => {
+            if (isLineOnEditMode(rowData)) {
+                const index = rowData.columnIndex - 1;
+                const Editor = COLUMNS_DEFINITIONS[index].editor;
+                if (Editor) {
+                    return (
+                        <div className={classes.tableCell}>
+                            <Editor
+                                key={rowData.dataKey + index}
+                                columnDefinition={COLUMNS_DEFINITIONS[index]}
+                                defaultValue={rowData.cellData}
+                                setColumnError={(k) => setColumnInError(k)}
+                                resetColumnError={(k) => resetColumnInError(k)}
+                                setter={(val) => handleEditCell(rowData, val)}
+                            />
+                        </div>
+                    );
+                }
+            }
+            return defaultCellRender(rowData);
+        },
+        [
+            COLUMNS_DEFINITIONS,
+            classes.tableCell,
+            defaultCellRender,
+            handleEditCell,
+            isLineOnEditMode,
+            resetColumnInError,
+            setColumnInError,
+        ]
+    );
+
+    const generateTableColumns = () => {
+        let tableColumns = Object.values(COLUMNS_DEFINITIONS).map((c) => {
+            if (c.editor) {
+                c.cellRenderer = editableCellRender;
+            }
+            return c;
+        });
+
+        tableColumns.unshift({
+            id: 'edit',
+            dataKey: 'edit',
+            label: '',
+            minWidth: 75,
+            locked: true,
+            editColumn: true,
+            cellRenderer: (rowData, rowIndex) =>
+                editCellRender(rowData, rowIndex),
+        });
+        return tableColumns;
+    };
+
+    const getCSVColumns = () => {
+        return [
+            intl.formatMessage({ id: 'Tap' }),
+            intl.formatMessage({ id: 'ImportFileResistance' }),
+            intl.formatMessage({ id: 'ImportFileReactance' }),
+            intl.formatMessage({ id: 'ImportFileConductance' }),
+            intl.formatMessage({ id: 'ImportFileSusceptance' }),
+            intl.formatMessage({ id: 'Ratio' }),
+        ];
+    };
+
+    const parseIntData = (data, defaultValue) => {
+        return isNaN(parseInt(data), 10) ? defaultValue : parseInt(data);
+    };
+
+    const handleImportTapRule = (selectedFile) => {
+        Papa.parse(selectedFile, {
+            header: true,
+            skipEmptyLines: true,
+            complete: function (results) {
+                let rows = results.data.map((val) => {
+                    return {
+                        key: results.data.indexOf(val),
+                        tap: val[intl.formatMessage({ id: 'Tap' })],
+                        resistance: parseIntData(
+                            val[
+                                intl.formatMessage({
```

Comment:
Why are the keys subject to internatilization/translation ?

Reply:
It's a requirement of the user story, the header of the CSV files must be translated

Reply:
This is OK, as discussed with the PO about the use case. No problem left :-)